packages/desktop-client/src/components/transactions/SelectedTransactionsButton.jsx

```diff
@@ -18,83 +20,129 @@ export function SelectedTransactionsButton({
   onCreateRule,
   onSetTransfer,
   onScheduleAction,
-  pushModal,
   showMakeTransfer,
+  onMakeAsSplitTransaction,
+  onMakeAsNonSplitTransactions,
 }) {
+  const dispatch = useDispatch();
   const selectedItems = useSelectedItems();
+  const selectedIds = useMemo(() => [...selectedItems], [selectedItems]);
 
   const types = useMemo(() => {
-    const items = [...selectedItems];
+    const items = selectedIds;
     return {
       preview: !!items.find(id => isPreviewId(id)),
       trans: !!items.find(id => !isPreviewId(id)),
     };
-  }, [selectedItems]);
+  }, [selectedIds]);
 
   const ambiguousDuplication = useMemo(() => {
-    const transactions = [...selectedItems].map(id => getTransaction(id));
+    const transactions = selectedIds.map(id => getTransaction(id));
 
     return transactions.some(t => t && t.is_child);
-  }, [selectedItems]);
+  }, [selectedIds, getTransaction]);
```

Comment:
<!-- Praises highlight something positive. Try to leave at least one of these comments per review (if it exists :^)  -->
**👏 praise:** ‏good catch!